tools/raisinbread_refresh.php

```diff
@@ -0,0 +1,358 @@
+ * manually change these values in a MySQL shell when they are not hosting a
+ * LORIS on localhost.
+ *
+ * Finally, the script runs tools/resetpassword.php so that the default admin
+ * password with the Raisinbread data set is not used.
+ *
+ * In order to prevent accidental data loss, the script prompts the user to
+ * manually type the exact name of the database which will be affected.
+ *
+ * In normal cases, the database connection will be established via the
+ * credentials in the config file. However, if the database tables have already
+ * been deleted then this will not work. In this case it is still possible to
+ * import Raisinbread data if the user has properly set up a MySQL configuration
+ * file and provides the name of their LORIS database.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Loris
+ * @author   John Saigle <john.saigle@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+
+$info = <<<INFO
+This script is used by LORIS developers to DELETE DATA and replace
+them with new test data.
+
+
+INFO;
+echo $info;
+
+echo "\e[0;31m*** Executing this script will result in the LOSS OF DATA! ***\e[0m\n";
+
+printHeader(
+    'Testing connection to database. No data will be affected at this point.'
+);
+try {
+    // Require generic libraries. In some cases, this can fail if this script
+    // has already dropped tables as the generic_includes.php requires a
+    // database connection.
+    include_once 'generic_includes.php';
+
+    $urlConfigSetting  = $config->getSetting('url');
+    $baseConfigSetting = $config->getSetting('base');
+    $hostConfigSetting = $config->getSetting('host');
+} catch (\DatabaseException $e) {
+    printWarning(
+        "Could not connect to the database in the Config file. " .
+        "It's possible that the LORIS tables have already been dropped."
+    );
+    echo 'Continue attempting to install Raisinbread database? (y/N)' . PHP_EOL;
+    $input = trim(fgets(STDIN));
+
+    if (mb_strtolower($input) !== 'y') {
+        die;
+    }
+}
+
+// Get database information from project's configuration file.
+$config   = $config ?? \NDB_Factory::singleton()->config();
+$dbInfo   = $config->getSettingFromXML('database');
+$dbname   = $dbInfo['database'];
+$host     = $dbInfo['host'];
+$username = $dbInfo['adminUser'];
+$password = $dbInfo['adminPassword'];
+
+if (! $config->getSetting('dev')['sandbox']) {
+    fwrite(
+        STDERR,
+        "Config file indicates that this is not a sandbox. Aborting to " .
+        "prevent accidental data loss." . PHP_EOL
+    );
+    exit(1);
+}
+
+echo <<<CONFIRMATION
+Please type the name of your database `$dbname` to confirm you wish to drop
+tables and import test data: 
+CONFIRMATION;
+
+$input = trim(fgets(STDIN));
+if ($input !== $dbname) {
+    die(printWarning('Input did not match database name. Exiting.'));
+}
+
+$mysqlCommand = <<<CMD
+mysql -A $dbname
```

Comment:
heredocs are appropriate for formatting multi-line strings which are getting directly displayed to the user. This does not meet that criteria. This should just be a normal string.

Also, the command arguments should be properly shell escaped.

Reply:
Alright I'll change these.